dask/utils.py

```diff
@@ -399,14 +402,21 @@ def __call__(self, arg):
             pass
         else:
             register()
-            return self(arg)  # recurse
+            return self  # recurse
```

Comment:
> I just want to confirm, either case works correct?

No, you need to recurse into the dispatch call here. If you don't, you return `self` (a `Dispatcher`), which is not what the user requested from calling `dispatch`. If the user is just calling the returned function then it will continue to recurse, but this isn't the intent.